ccm_web/server/src/api/api.create.section.handler.ts

```diff
@@ -0,0 +1,57 @@
+import CanvasRequestor from '@kth/canvas-api'
+import baseLogger from '../logger'
+import { CreateSectionResponseData, CreateSectionReturnResponse, CanvasSectionBase, CreateSectionsResponseObject, APIErrorData, handleAPIError } from './api.interfaces'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+export class CreateSectionApiHandler {
+  constructor (private readonly requestor: CanvasRequestor,
+    private readonly sections: string[],
+    private readonly courseId: number) {}
+
+  async apiCreateSection (sectionName: string): Promise<CreateSectionsResponseObject> {
+    try {
+      const endpoint = `courses/${this.courseId}/sections`
+      const method = 'POST'
+      const requestBody = { course_section: { name: sectionName } }
+      logger.debug(`Sending request to Canvas - Endpoint: ${endpoint}; Method: ${method}; Body: ${JSON.stringify(requestBody)}`)
+      const response = await this.requestor.requestUrl<CanvasSectionBase>(endpoint, method, requestBody)
+      return { statusCode: response.statusCode, message: response.statusMessage as string, sectionName: response.body.name }
+    } catch (error) {
+      const errResponse: APIErrorData = handleAPIError(error)
+      logger.error(`Request to create section ${sectionName} failed with StatusCode: ${errResponse.statusCode} and StatusMessage: ${errResponse.message} `)
+      return { statusCode: errResponse.statusCode, message: errResponse.message, sectionName: sectionName }
+    }
+  }
+
+  makeReturnResponseCreateSections (sectionsReturnRes: CreateSectionsResponseObject[]): CreateSectionReturnResponse {
+    const sectionsDataStore: CreateSectionResponseData = { createdSections: 0, givenSections: this.sections.length, statusCode: [], error: {} }
+    for (const section of sectionsReturnRes) {
+      const { statusCode, message, sectionName } = section
+      if (statusCode === 200 || statusCode === 201) {
+        sectionsDataStore.createdSections++
+      } else {
+        sectionsDataStore.error[sectionName] = message
+        sectionsDataStore.statusCode.push(statusCode)
+      }
+    }
+
+    if (sectionsDataStore.createdSections === sectionsDataStore.givenSections) {
+      return { statusCode: 201, message: { section: { success: true } } }
+    } else {
+      return {
+        statusCode: Math.max(...[...new Set(sectionsDataStore.statusCode)]),
+        message: { section: { success: false, error: sectionsDataStore.error } }
+      }
```

Comment:
I'm thinking we should come up with a better structure or way of doing this. It doesn't make sense to me to use the max status code; either we should keep track of each irregular status code and tie them to particular errors, or we should provide a more generic one explicitly.

Storing nested data under `message` feels clunky. If we want to give the client more data, then we might want to consider providing an alternative error data structure for these bulk processes.